tf_yarn/pytorch/tasks/worker.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import os
+import sys
+
+import torch
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.utils.data.distributed import DistributedSampler
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import skein
+
+from tf_yarn._task_commons import (
+    setup_logging, get_task_description, _get_experiment, _get_cluster_tasks
+)
+from tf_yarn import _internal, event
+from tf_yarn.pytorch.experiment import PytorchExperiment, DataLoaderArgs
+setup_logging()
+
+
+_logger = logging.getLogger(__name__)
+
+MASTER_ADDR = "MASTER_ADDR"
+MASTER_PORT = "MASTER_PORT"
+PYTORCH_DPP_RANK = "PYTORCH_DPP_RANK"
+
+
+def _log_sys_info() -> None:
+    _logger.info(f"Python {sys.version}")
+    _logger.info(f"Skein {skein.__version__}")
+    _logger.info(f"Pytorch {torch.__version__}")
+
+
+def _create_dataloader(
+    dataset: torch.utils.data.Dataset, dataloader_args: DataLoaderArgs
+) -> torch.utils.data.DataLoader:
+    sampler: DistributedSampler = DistributedSampler(
+        dataset, shuffle=dataloader_args.shuffle
+    )
+    return torch.utils.data.DataLoader(
+        dataset, sampler=sampler, batch_size=dataloader_args.batch_size,
+        num_workers=dataloader_args.num_workers, pin_memory=dataloader_args.pin_memory,
+        drop_last=dataloader_args.drop_last, timeout=dataloader_args.timeout,
+        prefetch_factor=dataloader_args.prefetch_factor, shuffle=False
+    )
+
+
+def _train(experiment: PytorchExperiment, device: int, rank: int, world_size: int) -> None:
+    _logger.info(f"[{os.getpid()}] device: {device}; rank: {rank}")
+    os.environ[PYTORCH_DPP_RANK] = str(rank)
+    dist.init_process_group("gloo", rank=rank, world_size=world_size)
+
+    model = experiment.model.to(device)
```

Comment:
Here device is an int, but shouldn't it be a string ? device could be something like cuda:0 no ?

Reply:
A device can be an int (device id) or string ({device name}:{device id}). If it is an int, the device is CUDA:{device id}. I describe the device by the device id directly because:
1. We only support GPUs for now
2. DDP needs the GPU device ID